packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -705,11 +705,9 @@ async function searchVectorStore<ChunkMetadata extends Jsonifiable = Jsonifiable
   query: string,
   params?: { limit?: number; score_threshold?: number }
 ): Promise<ScoredChunk<ChunkMetadata>[]> {
-  const scoredLcDocs = await vectorStore.similaritySearchWithScore(
-    query,
-    params?.limit ?? defaultLangchainChunkLimit,
-    params
-  );
+  const k = params?.limit ?? defaultLangchainChunkLimit;
+  delete params?.limit;
+  const scoredLcDocs = await vectorStore.similaritySearchWithScore(query, k, params);
```

Comment:
@mdepinet I feel like this was a bug, but I defer that to you. The issue was that `limit` was being passed as a `filter` to Pinecone and I kept getting an empty response. It took me too long to figure out the issue was on our side, not in Pinecone/LangChain 😅

Reply:
I would tend to do this to avoid mutating `params`, since someone else in this function could expect `limit` to still be defined.

```suggestion
  const scoredLcDocs = await vectorStore.similaritySearchWithScore(query, k, _.omit(params, 'limit'));
```

Or better yet, explicitly pass the known params we want to pass to pinecone, rather than letting everything through except known bad params:

```suggestion
  const scoredLcDocs = await vectorStore.similaritySearchWithScore(query, k, _.pick(params, 'score_threshold'));
```

Reply:
Not sure what should and should not be included, so I'm just gonna `omit` the `limit` parameter for now.